src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_formatter.rs

```diff
@@ -114,6 +117,7 @@ impl ElasticSearchOpenSearchFormatter {
     }
 
     pub fn convert_chunk(&self, chunk: StreamChunk) -> Result<Vec<BuildBulkPara>> {
+        let mut update_map: HashMap<String, RowRef<'_>> = HashMap::new();
```

Comment:
I think this map is unnecessary because `UpdateDelete` and `UpdateInsert` must be adjacent to each other so we only need to cache one row for the check.